homeassistant/components/tautulli/sensor.py

```diff
@@ -53,12 +65,183 @@
     }
 )
 
-SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+
+def get_top_stats(
+    home_stats: PyTautulliApiHomeStats, activity: PyTautulliApiActivity, key: str
+) -> str | None:
+    """Get top statistics."""
+    value = None
+    for stat in home_stats:
+        if stat.rows and stat.stat_id == key:
+            value = stat.rows[0].title
+        elif stat.rows and stat.stat_id == "top_users" and key == ATTR_TOP_USER:
+            value = stat.rows[0].user
+    return value
+
+
+@dataclass
+class TautulliSensorEntityMixin:
+    """Mixin for Tautulli sensor."""
+
+    value_fn: Callable[[PyTautulliApiHomeStats, PyTautulliApiActivity, str], StateType]
+
+
+@dataclass
+class TautulliSensorEntityDescription(
+    SensorEntityDescription, TautulliSensorEntityMixin
+):
+    """Describes a Tautulli sensor."""
+
+
+SENSOR_TYPES: tuple[TautulliSensorEntityDescription, ...] = (
+    TautulliSensorEntityDescription(
         icon="mdi:plex",
         key="watching_count",
         name="Tautulli",
         native_unit_of_measurement="Watching",
+        value_fn=lambda home_stats, activity, _: activity.stream_count or 0,
```

Comment:
I dont think we should add a default to any of the value_fn keys.
`None` is a valid state, and one that makes sense for all these.

Reply:
Made it `Any` because of mypy

Reply:
Any sounds incorrect / an escape hatch instead of a solution. Why is that needed?

Reply:
It seems that mypy is not seeing the typing after `lambda` is declared hence why it's currently set to `Any`.

Reply:
That is probably because the upstream library isn't typed.
So, multiple approaches are possible here; for example, cast the results here, or even better: Add typing to the upstream library.

Reply:
There is typing there, its just wrong.
I want to rewrite that soon anyway and will fix it then.
Lets cast here, and we can do it better later.